mllm/models/minicpm4/modeling_minicpm4.hpp

```
@@ -0,0 +1,393 @@
+#include "mllm/nn/Module.hpp"
+#include "mllm/nn/Nn.hpp"
+#include "mllm/nn/Functional.hpp"
+#include "mllm/nn/lmcache/StaticCache.hpp"
+#include "mllm/models/minicpm4/configuration_minicpm4.hpp"
+#include "mllm/utils/Enumerate.hpp"
+#include "mllm/models/ARGeneration.hpp"
+#include <cmath>
+
+namespace mllm::models::minicpm4 {
+
+inline auto makeRoPEInvFreqWithLongRoPE(int output_dim, float rope_theta, const MiniCPM4Config& cfg, int seq_len) -> Tensor {
```

Clang (14.0.6)

[error] 4 adjacent parameters of 'makeRoPEInvFreqWithLongRoPE' of convertible types are easily swapped by mistake

(bugprone-easily-swappable-parameters,-warnings-as-errors)